CODE_OF_CONDUCT.md

```diff
@@ -5,33 +5,17 @@ See commit history to see our changes.
 ## Our Pledge
 
 We as members, contributors, and leaders pledge to make participation in our
-community a harassment-free experience for everyone, regardless of age, body
-size, visible or invisible disability, ethnicity, sex characteristics, gender
-identity and expression, level of experience, education, socio-economic status,
-nationality, personal appearance, race, caste, color, religion, or sexual
-identity and orientation.
```

Comment:
Removing this kinda breaks the whole point of having a CoC

Reply:
No it doesn't. I just removed the repetition. "everyone" implicitly includes all of those classifiers.

Reply:
I think it's better to keep that

Reply:
I think it's better to get to the point instead of listing pointless classifiers after already saying "everyone".

Reply:
This part was about being actually inclusive. Your "everyone" is not *everyone's* "everyone"

Reply:
i agree with lenny, repeating is stupid, everyone is everyone, having a paragraph thats not exclusive anyways just makes pepole want to read it less

Reply:
> This part was about being actually inclusive. Your "everyone" is not _everyone's_ "everyone"

this

Reply:
Having these included makes people who are discriminated against for these specific reasons feel safe. A blanket "everyone" does not. This is coming from a person who has been discriminated against on the basis of age, invisible disabilities, nationality, personal appearance, religion (specifically a lack thereof), and sexual orientation. I'd rather these be left in.

The other changes are fine.

Reply:
> Having these included makes people who are discriminated against for these specific reasons feel safe. A blanket "everyone" does not. This is coming from a person who has been discriminated against on the basis of age, invisible disabilities, nationality, personal appearance, religion (specifically a lack thereof), and sexual orientation. I'd rather these be left in.
> 
> The other changes are fine.

I don't really understand how "everyone" is different from "everyone regardless of x, y, z" because to me they seem to mean the same thing since "everyone" already doesn't exclude anyone. What about it makes it feel more safe? To me, it seems like both of them are against all types of discrimination so they would both feel safe against discrimination.

Reply:
From Contributor Covenant FAQ (last one)

> # Contributor Covenant explicitly lists a set of protected classes; does this make it acceptable to discriminate or make others feel unwelcome based on other factors?
> No. Contributor Covenant explicitly lists protected classes for many reasons, such as reminding people to give them appropriate consideration, and assuring people in those protected classes that they are welcome. However, this is not an invitation for rules lawyers to seek loopholes, or to discriminate against others or make people feel unwelcome based on criteria not listed here. (With the notable caveat that those who discriminate or make others feel unwelcome are themselves not welcome.)

Reply:
> From Contributor Covenant FAQ (last one)
> 
> > # Contributor Covenant explicitly lists a set of protected classes; does this make it acceptable to discriminate or make others feel unwelcome based on other factors?
> > No. Contributor Covenant explicitly lists protected classes for many reasons, such as reminding people to give them appropriate consideration, and assuring people in those protected classes that they are welcome. However, this is not an invitation for rules lawyers to seek loopholes, or to discriminate against others or make people feel unwelcome based on criteria not listed here. (With the notable caveat that those who discriminate or make others feel unwelcome are themselves not welcome.)

I just don't see how it provides any more assurance than saying "everyone" because "everyone" still doesn't exclude anyone.

Reply:
I guess if it helps keep it.

Reply:
You are looking at this as if it was a math problem. The idea is to explicitly include these groups of people to reassure them, that they are welcome.

Reply:
I really don't see why they would feel *unwelcome* when seeing "We as members, contributors, and leaders pledge to make participation in our community a harassment-free experience for everyone."

Reply:
The Contributor Covenant is a politically-motivated document that was pushed onto many open source projects - and with much controversy - since its inception circa 2018. One serious question that should be asked is, does the project need this document at all?

It's certainly overkill if all you want is for all contributors to treat others professionally and courteously. Just say that.

Reply:
> The Contributor Covenant is a politically-motivated document 

Explain how

> that was pushed onto many open source projects - and with much controversy - since its inception circa 2018. 

How was it “pushed”? Are you implying it or similar documents were forcefully imposed?

> One serious question that should be asked is, does the project need this document at all?
> 
> It's certainly overkill if all you want is for all contributors to treat others professionally and courteously. Just say that.

People can and will toe the line on purpose to be assholes or to spread bigotry. The point of these sorts of documents is to make very clear what is and is not considered “professional and courteous”. Almost any project the size of PolyMC or larger needs such a document.

Reply:
@LennyMcLennington Here. Here is the section of the diff where several other people are saying the exact same thing I am. For some reason this didn’t show up when I was logged out so presumably not visible on your end. Reposting.

 Now, stop playing dumb.

Reply:
I can see both sides here, but getting political about it really says more about the person arguing, more than anything else.
Subscribe to the MEGA code of merit: Make cool shit till you die.
Stay safe.

Reply:
> Explain how

A bit lengthy to get into here so I'll try to summarize; I paid attention when it first appeared around five years ago and tore its way across software projects.

It is a manifesto distracting from development in favor of critical social justice activism. The creator of the document concurrently advocates for so-called "Post-Meritocracy." It was extremely divisive and drove talent away from projects who found the resulting conditions of walking on eggshells untenable to work in. 

> How was it “pushed”? Are you implying it or similar documents were forcefully imposed?

Yes, there was a considerable push by activists who insisted on its adoption and who opened requests on myriad projects. One of the few notable projects to reject it at the time because they saw it for the politicization of their project that it was, was SQLite.

> People can and will toe the line on purpose to be assholes or to spread bigotry. The point of these sorts of documents is to make very clear what is and is not considered “professional and courteous”. Almost any project the size of PolyMC or larger needs such a document.

Respectfully, I think the degree to which this is actually an issue is overestimated. If someone causes problems then you moderate proportionally. Bad faith actors won't care what you have written in your document, and good faith actors know how to conduct themselves appropriately sans condescension.